system/micd.py

```diff
@@ -12,6 +12,15 @@
 SAMPLE_RATE = 44100
 SAMPLE_BUFFER = 4096 # (approx 100ms)
 
+def precompute_a_weighting():
+  # Calculate the A-weighting filter
+  # https://en.wikipedia.org/wiki/A-weighting
+  freqs = np.fft.fftfreq(FFT_SAMPLES, d=1 / SAMPLE_RATE)
+  A = 12194 ** 2 * freqs ** 4 / ((freqs ** 2 + 20.6 ** 2) * (freqs ** 2 + 12194 ** 2) * np.sqrt((freqs ** 2 + 107.7 ** 2) * (freqs ** 2 + 737.9 ** 2)))
+  return A / np.max(A)
+
+
+A_WEIGHTING = precompute_a_weighting()
```

Comment:
is caching not better in your opinion?

Reply:
The filter is computed once and reused, so a global variable works well. 
Using a global variable for caching is likely the better approach here for the following reason:
1. The result doesn’t change with input, so no need for dynamic caching.
2. Global variables are simple,straightforward and avoid the overhead of cache decorators when results are fixed.

Reply:
> caching seems cleaner to me

I agree, caching does offer a cleaner approach. If you think it's the way to go, let's use that.